core/src/main/java/jenkins/mvn/SettingsProviderHelper.java

```diff
@@ -0,0 +1,117 @@
+
+    public static boolean ABSOlUTE_PATH_BACKWARD_COMPATIBILITY = Boolean.valueOf(System.getProperty("jenkins.mvn.SettingsProvider.absolutePathBackwardCompatibility"));
+
+    /**
+     * Provide the {@link FilePath} of the given plain text {@code path} for the desired Maven settings.xml file.
+     *
+     * @param build       the build to provide the settings for
+     * @param workspace the workspace in which the build takes place
+     * @param listener the listener of this given build
+     * @return the filepath to the provided file. {@code null} if no settings will be provided.
+     * @throws IOException exception accessing to the file system
+     * @throws InterruptedException occurs when the build is interrupted
+     */
+    @CheckForNull
+    public static FilePath supplySettings(String path, Run<?, ?> build, FilePath workspace, TaskListener listener) throws IOException, InterruptedException {
+        if (StringUtils.isEmpty(path)) {
+            return null;
+        }
+
+        try {
+            EnvVars env = build.getEnvironment(listener);
+
+            FilePath result;
+            if (build instanceof AbstractBuild) {
+                AbstractBuild abstractBuild = (AbstractBuild) build;
+                String targetPath = Util.replaceMacro(path, abstractBuild.getBuildVariableResolver());
+                targetPath = env.expand(targetPath);
+                if (IOUtils.isAbsolute(targetPath) && ABSOlUTE_PATH_BACKWARD_COMPATIBILITY) {
+                    File settingsFile = new File(targetPath);
+                    checkFileIsaMavenSettingsFile(settingsFile);
+                    return new FilePath(settingsFile);
+                } else {
+                    FilePath mrSettings = abstractBuild.getModuleRoot().child(targetPath);
+                    FilePath wsSettings = abstractBuild.getWorkspace().child(targetPath);
+                    if (wsSettings.exists()) {
+                        result = wsSettings;
+                    } else if (mrSettings.exists()) {
+                        result = mrSettings;
+                    } else {
+                        throw new FileNotFoundException(targetPath);
+                    }
+                }
+            } else {
+                String targetPath = env.expand(path);
+                result = workspace.child(targetPath);
+            }
+            return result;
+        } catch (IOException e) {
+            throw new IOException(
+                    "Failed to prepare Maven (global) settings.xml with path '" + path + "' for " + build +
+                            " in workspace " + workspace, e);
+        } catch (InterruptedException e) {
+            throw new IOException(
+                    "Failed to prepare Maven (global) settings.xml with path '" + path + "' for " + build +
+                            " in workspace " + workspace, e);
+        } catch (Exception e) {
+            throw new IllegalStateException(
```

Comment:
Do not use Runtime Exceptions in new code unless required 🐛 
In your case you can just replace all 3 catches by...

```
catch (InterruptedException | IOException | RuntimeException e) {
 +            throw new IOException(
 +                    "Failed to prepare Maven (global) settings.xml with path '" + path + "' for " + build +
 +                            " in workspace " + workspace, e);
 +        }
```